### PR TITLE
extend benchmark to make the db-sync mode and node-profiler configurable

### PR DESCRIPTION
**Pull request by arvidn**
example output from the node profiler:

image

### CODE REVIEW

**Review by 1-justinstarr — COMMENTED** [review]
running

File: tools/run_benchmark.sh [hunk 1]
```diff
@@ -2,13 +2,15 @@
 
 run_benchmark() {
    # shellcheck disable=SC2086
-   python ./tools/test_full_sync.py run $3 --profile --test-constants "$1" &
+   python ./tools/test_full_sync.py run $3 --profile --node-profiler --test-constants "$1" &
```
Comment (arvidn) [inline tools/run_benchmark.sh:5]: it just occurred to me that combining `--profile` and `--node-profiler` here will cause two `cProfile` instances to be active at the same time. Is that supported by python? I don't recall anything stating it isn't..

Comment (mariano54) [inline tools/run_benchmark.sh:5]: Is there any reduced performance from this?

**Review by mariano54 — DISMISSED** [review]

**Review by trepca — DISMISSED** [review]

**Review by trepca — APPROVED** [review]
lgtm

**Review by wjblanke — APPROVED** [review]
aok

**Pull request merged**